### PR TITLE
[HttpFoundation][FrameworkBundle] Add `BinaryFileResponse::setXSendfileHeader()` and corresponding config option to help configure upload acceleration

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 7.2
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Issues        | -
| License       | MIT

Follows https://github.com/symfony/symfony/pull/58161#discussion_r1742107766 (and embeds the corresponding commit for now)

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ CHANGELOG
  * Deprecate making `cache.app` adapter taggable, use the `cache.app.taggable` adapter instead
  * Enable `json_decode_detailed_errors` in the default serializer context in debug mode by default when `seld/jsonlint` is installed
  * Register `Symfony\Component\Serializer\NameConverter\SnakeCaseToCamelCaseNameConverter` as a service named `serializer.name_converter.snake_case_to_camel_case` if available
+ * Add support for `SYMFONY_TRUSTED_PROXIES`, `SYMFONY_TRUSTED_HEADERS`, `SYMFONY_TRUST_X_SENDFILE_TYPE_HEADER` and `SYMFONY_TRUSTED_HOSTS` env vars
```
Comment (OskarStark) [inline src/Symfony/Bundle/FrameworkBundle/CHANGELOG.md:15]: Does this really belong to this PR?

Comment (nicolas-grekas) [reply src/Symfony/Bundle/FrameworkBundle/CHANGELOG.md:15]: See PR description ;)
This belongs to https://github.com/symfony/symfony/pull/58161

Comment (OskarStark) [reply src/Symfony/Bundle/FrameworkBundle/CHANGELOG.md:15]: Yes, my understanding was, that you just cherry picked it, nvm

**Comment by nicolas-grekas** [comment]
Let me withdraw this PR: X-Senfile-Type is coming from the RoR world, and requires the server to send an X-Accel-Mapping header anyway so there is always something to configure on the server-side (at least for nginx)

See https://github.com/rack/rack/blob/main/lib/rack/sendfile.rb and https://mattbrictson.com/blog/accelerated-rails-downloads for more on the topic.

**Comment by nicolas-grekas** [comment]
See also https://github.com/symfony/symfony/pull/58181

**Pull request closed without merging**